### ISSUE TITLE
Undefined LEVELDB_IS_BIG_ENDIAN

### ISSUE THREAD

**Issue by studyCheng** [google/leveldb#572]
port/port_posix.h 50: unfefined LEVELDB_IS_BIG_ENDIAN

**Comment by Hadesonly** [comment]
same question, need help.

**Comment by studyCheng** [comment]
I define the LEVELDB_IS_BIG_ENDIAN.
#ifndef LEVELDB_IS_BIG_ENDIAN
#define LEVELDB_IS_BIG_ENDIAN 0
#endif

I can make success, but I don’t know if it’s right or wrong

**Comment by Hadesonly** [comment]
It can make by define the LEVELDB_IS_BIG_ENDIAN.
Then it doesn't have libleveldb.so*, so it can't work.

**Comment by studyCheng** [comment]
Although it doesn't have libleveldb.so*, it has libleveldb.a. you can use libleveldb.a to work.

**Comment by Ged0** [comment]
It also happened in CentOS Linux release 7.4.
```
mkdir build
cd build
cmake ..
make
```
error info
```
[  1%] Building CXX object CMakeFiles/leveldb_port_posix.dir/port/port_posix.cc.o
In file included from /root/github/leveldb/port/port_posix.cc:5:0:
/root/github/leveldb/port/port_posix.h:50:36: error: ‘LEVELDB_IS_BIG_ENDIAN’ was not declared in this scope
 static const bool kLittleEndian = !LEVELDB_IS_BIG_ENDIAN;
                                    ^
make[2]: *** [CMakeFiles/leveldb_port_posix.dir/port/port_posix.cc.o] Error 1
make[1]: *** [CMakeFiles/leveldb_port_posix.dir/all] Error 2
make: *** [all] Error 2
```
I find #define LEVELDB_IS_BIG_ENDIAN 0 in "./build/include/port/port_config.h".
Did I make something wrong when build leveldb?

**Comment by pwnall** [comment]
Everyone: This project only supports out-of-tree builds. So you have to do something like

```bash
mkdir build && cd build
cmake .. && cmake --build .
```

These instructions will be added to the README soon. You should be able to replace the 2nd `cmake` invocation with `make all` if you wish to. 

If you're doing an out-of-tree build, and using CMake >= 3.9, please check for a `LEVELDB_IS_BIG_ENDIAN ` definition in `include/port/port_config.h` in your CMake build directory. Please add a comment describing you find, and your OS / distribution version.

**Comment by pwnall** [comment]
@Hadesonly I think you want `-DBUILD_SHARED_LIBS=1` on your CMake command line. See https://cmake.org/cmake/help/v3.9/variable/BUILD_SHARED_LIBS.html

**Comment by olt** [comment]
@pwnall I'm hitting this issue as well. I think the issue is that `port_config.h` is not included with older compilers and `LEVELDB_HAS_PORT_CONFIG_H` is not set (by cmake?):
https://github.com/google/leveldb/blob/6fa45666703add49f77652b2eadd874d49aedaf6/port/port_posix.h#L10-L12

**Comment by pwnall** [comment]
@olt Thank you very much for debugging! I think you're right!

Can you please try https://github.com/pwnall/leveldb/tree/no-has-include and see if that fixes the problem for you?

**Comment by olt** [comment]
@pwnall Yes, it's now working with the default compiler on Ubuntu 14.04 (travis-ci). Thanks!

**Comment by pwnall** [comment]
@olt Once again, thank you! The master branch should now build as well.
